### ISSUE TITLE
Verify map teleport behaviors

### ISSUE THREAD

**Issue by fmatthew5876** [EasyRPG/Player#2275]
When looking at RE code, I found a few surprising things. Writing these down to be verified which ones we do and which we need to fix.

Function address in dynrpg binary 004A4488

All of these behaviors occur during a map teleport

- [ ] If the hero is exiting a vehicle, none of the below occurs (does it happen somewhere else??)

- [ ] The erase transition is not performed if the hero character is not enabled (impossible in practice, but code is there)
    * enabled flag is true (not possible to be false, but it's in the code.
    * transition type is not none!

- [ ] The hero's stepping animation is not reset if the screen is never erased! This means if the screen erase transition is None, and the screen is not already erase, the stepping animation will not change.

In other words, if the erase transition is disabled, character animation steps are not reset!

- [ ] Screen flash, screen animations, and pictures are only cleared if the map changes

- [ ] After setting hero direction, diagonal facing is reset

- [ ] All events processed flags are reset

-  [ ]
